In [ ]:
# GBA 462 Lab 6

# Liuyi Ye / Xinyi Wang

# 10/17/2023



 **Case Study: Employee tenure and store performance (Similar to HW5)**

You run a family business that sell fresh farm goods and you have opened serveral stores in the local area. To boost the financial
performance, you want to have a better idea of the relationship of employee retention and store profit. A file named **lab_data.csv**
contains data you have regarding the store-level characteristics in the following 9 aspects: Profit, MTenure (on average, how
many months have a manager been with this store), CTenure (on average, how many months have a crew member been with this
store), Comp (number of competitiors per 100,000 people in the city), Pop (city population), Visiblity (visibility rating from 1 to
5, with 5 being the highest), PedCount (pedestrian foot traffic rating from 1 to 5, with 5 being the highest), Hours24 (indicator
equal to 1 if the store opens 24 hours), Res (indicator equal to 1 if the city is largely a residential, as opposed to industrial, area)



1.Run a regression of *Profit* on all the explanatory variables above. Based on your regression results, intepret the coefficient of *MTenure* and *CTenure* and evaluate their statistical significance at the 1% significance level. What about Res?



In [ ]:
# import packages
import pandas as pd
import numpy as np
import statsmodels.formula.api as smf
from scipy import optimize


In [ ]:
lab_data = pd.read_csv("lab_data.csv")
lab_data.head()

Store     Sales       Profit   MTenure    CTenure    Pop      Comp  \
0      1  5301.470  1766.760000   0.80000  25.804930   7535  3.637254   
1      2  8099.370  2826.713333  87.02219   7.636550   8630  5.506221   
2      3  5499.605  1484.900000  24.68854   6.026694   9695  5.843066   
3      4  5269.300  1400.813333   0.80000   6.371663   2797  5.530130   
4      5  6139.205  2003.200000   4.67737   7.866530  20335  2.146773   

   Visibility  PedCount  Res  Hours24  
0           4         3    1        1  
1           5         3    1        1  
2           4         3    1        1  
3           5         2    1        1  
4           3         5    0        1

In [ ]:
lab_data.shape

(55, 11)

In [ ]:
# When you do your homework, you can use the following code chunk to read data.
hw_data = pd.read_csv("http://paulellickson.com/ClassData/Store24.csv")
hw_data.head()

Store    Sales  Profit   MTenure    CTenure    Pop      Comp  Visibility  \
0      1  1060294  265014   0.00000  24.804930   7535  2.797888           3   
1      2  1619874  424007  86.22219   6.636550   8630  4.235555           4   
2      3  1099921  222735  23.88854   5.026694   9695  4.494666           3   
3      4  1053860  210122   0.00000   5.371663   2797  4.253946           4   
4      5  1227841  300480   3.87737   6.866530  20335  1.651364           2   

   PedCount  Res  Hours24  
0         3    1        1  
1         3    1        1  
2         3    1        1  
3         2    1        1  
4         5    0        1

basic data investigation and simple regressions

In [ ]:
lab_data.describe()

Store         Sales       Profit     MTenure     CTenure  \
count  55.00000     55.000000    55.000000   55.000000   55.000000   
mean   28.00000   6186.785909  1881.147394   47.025538   15.779914   
std    16.02082   1540.926110   588.315670   59.724009   19.558671   
min     1.00000   3721.055000   982.180000    0.800000    2.347023   
25%    14.50000   4998.060000  1437.346667    7.470391    6.314168   
50%    28.00000   5857.455000  1782.360000   24.918560    9.246407   
75%    41.50000   7029.805000  2236.133333   54.360280   18.215610   
max    55.00000  10565.445000  3175.700000  278.787700  115.151900   

                Pop       Comp  Visibility   PedCount        Res    Hours24  
count     55.000000  55.000000   55.000000  55.000000  55.000000  55.000000  
mean   10515.236364   5.026122    4.090909   3.054545   0.963636   0.854545  
std     6092.423558   1.830252    0.776333   0.950987   0.188919   0.355808  
min     1116.000000   2.146773    3.000000   1.000000   0.000000   0.000000  
25%     6714.500000   4.121963    4.000000   2.000000   1.000000   1.000000  
50%     9285.000000   4.749733    4.000000   3.000000   1.000000   1.000000  
75%    14565.000000   5.602055    5.000000   4.000000   1.000000   1.000000  
max    26519.000000  14.466244    6.000000   5.000000   1.000000   1.000000

In [ ]:
reg=smf.ols(formula='Profit~MTenure+CTenure+Pop+Comp+Visibility+PedCount+Res+Hours24', data=lab_data)
est = reg.fit()
print(est.summary())

                            OLS Regression Results                            
Dep. Variable:                 Profit   R-squared:                       0.666
Model:                            OLS   Adj. R-squared:                  0.608
Method:                 Least Squares   F-statistic:                     11.47
Date:                Wed, 18 Oct 2023   Prob (F-statistic):           9.01e-09
Time:                        17:39:51   Log-Likelihood:                -398.12
No. Observations:                  55   AIC:                             814.2
Df Residuals:                      46   BIC:                             832.3
Df Model:                           8                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept    145.3391    560.154      0.259      0.7



*   The coefficient of *MTenure* is roughly 5.5, implying that an additional month of manager tenure is associated with an expected increase in (yearly) profit of $5.5 holding other things constant; *MTenure *is statistically significant at the 1% level (p-value < 0.01);

*   The interpretation of *CTenure* is pretty similar to *MTenure*. *CTenure* is not significant at the 1% level. However, it is significant at the 10% level (0.05 < p-value < 0.1)

* The coefficient of *Res* implies that, compared with stores located at the industrial area, stores located at the residential area would have a $709 higher (yearly) profit holding other things constant. It is not significant at the 1% level, but significant at the 5% level (0.01 < p-value < 0.05)


2.Before the regression analysis, you were considering contracting part of managers’ salaries on their store crew retention (i.e., if
employees stay with the store for longer, the managers of the store get higher pay). Based on your results, what is your estimate of the impact of a **1.412-month** crew tenure increase on store profits? How would you proceed with estimate to better evaluate your ealier contracting plan?


In [ ]:
change_profit = est.params['CTenure'] * 1.412
print("Expected profit change:" , change_profit)

Expected profit change: 7.500617691715291


In [ ]:
ci = 1.412 * est.conf_int(alpha=0.05).loc[['CTenure']]
print("Confidence interval for change: ", ci.values[0])

Confidence interval for change:  [-1.11733596 16.11857135]


     With 95% probability we cannot rule out the change in profit is as low as -\$1.1 and as high as \$16.1



*   How to proceed from here? - Discussion based on different case scenarios





3.Construct a 95% CI for the effect of Comp on profits and interpret the CI.


In [ ]:
ci = est.conf_int(alpha=0.05).loc[['Comp']]
print("Confidence interval for Comp:", ci.values[0])

Confidence interval for Comp: [-184.78925975  -63.19043329]


*  The 95% CI for the impact of Comp on profits is (-184.8, -63.2)
*  With 95% probability, we believe the true value of the population parameter for *Comp* will be in the range (-184.8, -63.2). In other words, the **trueeffect** of a unit increase in *Comp* on yearly profits is between \$-184.8 and \$-63.2.


4.Among the explanatory variables, *Comp, Pop, Visiblity, PedCount, Hours24* and *Res* are location-based. What are the roles of these location-based variables? Do you expect them to have a certain sign? Rerun the earlier regression with these location-based variables dropped. What does the results tell you?


In [ ]:
est = smf.ols(formula='Profit ~ MTenure + CTenure', data=lab_data).fit()
print(est.summary())

                            OLS Regression Results                            
Dep. Variable:                 Profit   R-squared:                       0.255
Model:                            OLS   Adj. R-squared:                  0.226
Method:                 Least Squares   F-statistic:                     8.881
Date:                Wed, 18 Oct 2023   Prob (F-statistic):           0.000481
Time:                        17:43:43   Log-Likelihood:                -420.21
No. Observations:                  55   AIC:                             846.4
Df Residuals:                      52   BIC:                             852.4
Df Model:                           2                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept   1596.6465    101.035     15.803      0.0

  * After dropping the location-based variables $\rightarrow$

  (1) Adj-R2 dropped from 0.6 to 0.23, worse model fit.

  (2) Omitting location-based variables causes omitted variable bias (point estimates of MTenure and CTenure change), also lowers estimate precision (e.g., CTenure’s standard error/t-stat/p-value)

  * These variables should definitely be included


5.How would you design the regression if the impact of *MTenure* and *CTenure* on *Profit* varies with the level of tenure? Run the regression and find the point where an additional month of MTenure leads to lower profit.

In [ ]:
form1='Profit~MTenure+np.power(MTenure,2)+CTenure+np.power(CTenure,2)+Pop+Comp+Visibility+PedCount+Res+Hours24'
est = smf.ols(formula=form1, data=lab_data).fit()
print(est.summary())


                            OLS Regression Results                            
Dep. Variable:                 Profit   R-squared:                       0.755
Model:                            OLS   Adj. R-squared:                  0.700
Method:                 Least Squares   F-statistic:                     13.58
Date:                Wed, 18 Oct 2023   Prob (F-statistic):           1.83e-10
Time:                        17:46:34   Log-Likelihood:                -389.58
No. Observations:                  55   AIC:                             801.2
Df Residuals:                      44   BIC:                             823.2
Df Model:                          10                                         
Covariance Type:            nonrobust                                         
                           coef    std err          t      P>|t|      [0.025      0.975]
----------------------------------------------------------------------------------------
Intercept               18.0682 

* We do find diminishing returns to MTenure and CTenure ! (any thoughts on why? What else do you observe from the regression output?)

* To find the point where the profit starts to decrease with MTenure , we just need to find the maximal point, because the profit first increases, then decreases.

Profit = 13.68 MTenure - 0.037 MTenure^2  + .... (terms unrelated to Mtenure)
- Take derivative with respect to *Mtenure*, 13.68 - 0.074MTenure = 0 $\rightarrow$ MTenure = 184 months

In [ ]:
#the other apporach

def f(x):
  return -(est.params['MTenure']*x + est.params['np.power(MTenure, 2)']*x**2)
result = optimize.minimize_scalar(f)
print(result.success) # check if solver was successful

True


In [ ]:
result

 message: 
          Optimization terminated successfully;
          The returned value satisfies the termination criteria
          (using xtol = 1.48e-08 )
 success: True
     fun: -1262.0890788092745
       x: 184.56502241107418
     nit: 4
    nfev: 10